pkg/tls/tls.go

```diff
@@ -139,14 +150,18 @@ type SDKCertGenerator struct {
 	KubeClient kubernetes.Interface
 }
 
-// GenerateCert returns a secret containing the TLS encryption key and cert,
-// a ConfigMap containing the CA Certificate and a Secret containing the CA key or it
-// returns a error incase something goes wrong.
-func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *v1.Service, config *CertConfig) (*v1.Secret, *v1.ConfigMap, *v1.Secret, error) {
+// GenerateCert returns a Secret containing the TLS encryption key and cert,
+// a ConfigMap containing the CA Certificate, and a Secret containing the CA key.
+// GenerateCert returns an error if generation fails.
+func (scg *SDKCertGenerator) GenerateCert(cr runtime.Object, service *corev1.Service, config *CertConfig) (*corev1.Secret, *corev1.ConfigMap, *corev1.Secret, error) {
 	if err := verifyConfig(config); err != nil {
 		return nil, nil, nil, err
 	}
 
+	var ownerRefs []metav1.OwnerReference
+	if service != nil {
+		ownerRefs = service.GetOwnerReferences()
```

Comment:
Can we assume that `service.OwnerReferences` can be applied to TLS objects?

Reply:
You mean if the `Service` should be the Owner of the TLS objects?

Reply:
Should the owners of `service` be the owners of TLS objects.

Reply:
Depends how long you want the lifecycle of the TLS objects to be? If I understand correctly the `Secret` that is passed in should be the owner, as generating the TLS objects depends on the Secret object that is passed to the `GenerateCert ` function. So I would say `Service` should be the owner, wdyt?